ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/MergeRewriter.java

```diff
@@ -217,13 +218,17 @@ public void appendWhenMatchedUpdateClause(MergeStatement.UpdateClause updateClau
     }
 
     protected void addValues(Table targetTable, String targetAlias, Map<String, String> newValues,
-                             List<String> values) {
+                             List<String> values, boolean aliasRhsExpr) {
       for (FieldSchema fieldSchema : targetTable.getCols()) {
+        String value = String.format("%s.%s", targetAlias, HiveUtils.unparseIdentifier(fieldSchema.getName(), conf));
         if (newValues.containsKey(fieldSchema.getName())) {
-          values.add(newValues.get(fieldSchema.getName()));
+          String rhsExp = newValues.get(fieldSchema.getName());
+          if (aliasRhsExpr){
+            rhsExp += String.format(" AS %s", value);
+          }
+          values.add(rhsExp);
```

Comment:
How about extracting this to method to get rid of the boolean flag `aliasRhsExpr` since UUIC it's value is implementation specific?
```
getNewValue(...) {
  return fieldSchema.getName();
}
```
and override it in `CopyOnWriteMergeWhenClauseSqlGenerator`
```
@Override
getNewValue(...) {
  String value = String.format("%s.%s", targetAlias, HiveUtils.unparseIdentifier(fieldSchema.getName(), conf));
  return String.format("%s AS %s", fieldSchema.getName(), value);
}
```

Reply:
ok

Reply:
+ added extra refactor

Reply:
How about 
```
MergeRwriter.addValues(...) {
      for (FieldSchema fieldSchema : targetTable.getCols()) {
        String quotedColumnName = String.format("%s.%s", targetAlias,
            HiveUtils.unparseIdentifier(fieldSchema.getName(), conf));
        if (newValues.containsKey(fieldSchema.getName())) {
          values.add(getValue(newValues.get(fieldSchema.getName()), quotedColumnName));
        } else {
          values.add(quotedColumnName);
        }
      }
}

protected String MergeRwriter.getValue(String newValue, String alias) {
  return newValue;
}
```
```
@Override
protected String CopyOnWriteMergeRewriter.getValue(String newValue, String alias) {
  return String.format("%s AS %s", newValue, alias);
}
```
every code part is on the place where it's suppose to be and the logic in `MergeRwriter.addValues` is less complex: `if (aliasRhsExpr){` and a method parameter is removed.

OR

If `it won't harm if we add alias in existing implementations` then why don't we add it always. It also means that the boolean parameter `aliasRhsExpr` can be removed.

Reply:
i don't see any benefit in this, `aliasRhsExpr` is an extra param, that is false in a base class and true in 1 of the children. btw, MergeRwriter - is a base class

Reply:
changed, however, I don't think that was a critical issue